File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/networking/ApiError.kt
package com.desktopkitchen.pos.networking

sealed class ApiError : Exception() {
    data class HttpError(val statusCode: Int, override val message: String) : ApiError()
    data class NetworkError(override val cause: Throwable) : ApiError() {
        override val message: String get() = cause.message ?: "Network error"
    }
    data class DecodingError(override val cause: Throwable) : ApiError() {
        override val message: String get() = cause.message ?: "Decoding error"
    }
    data object Unknown : ApiError() {
        override val message: String get() = "Unknown error"
    }
}

File: apps/pos/native-windows/DesktopKitchenPOS/Networking/ApiError.cs
namespace DesktopKitchenPOS.Networking;

public class ApiException : Exception
{
    public int? StatusCode { get; }

    public ApiException(string message, int? statusCode = null)
        : base(message)
    {
        StatusCode = statusCode;
    }

    public ApiException(string message, Exception inner, int? statusCode = null)
        : base(message, inner)
    {
        StatusCode = statusCode;
    }
}
